### PR TITLE
Feature/route finder single route

### LINKED ISSUE #567: [M2] Make changes to route finder.
Specs: https://github.com/watercompany/skywire-specs/blob/master/proposals/router2.md#route-finder-changes

### PR DESCRIPTION
**Pull request by ivcosla**
Closes #567

### CODE REVIEW

File: pkg/route-finder/client/client.go [hunk 1]
```diff
@@ -72,23 +70,21 @@ func NewHTTP(addr string, apiTimeout time.Duration) Client {
...
-// the given maxHops as well as the reverse routes from destiny to source.
-func (c *apiClient) PairedRoutes(source, destiny cipher.PubKey, minHops, maxHops uint16) ([]routing.Route, []routing.Route, error) {
-	requestBody := &GetRoutesRequest{
-		SrcPK:   source,
-		DstPK:   destiny,
-		MinHops: minHops,
-		MaxHops: maxHops,
+// FindRoutes returns routes from source skywire visor to destiny, that has at least the given minHops and as much
+// the given maxHops.
+func (c *apiClient) FindRoutes(ctx context.Context, rts [][2]cipher.PubKey, opts *RouteOptions) ([][]routing.Route, error) {
```
Comment (evanlinjin) [inline pkg/route-finder/client/client.go]: Have the following changes:
- Create a new type `type Path []routing.Hop`
- Use the following as the output: `map[[2]cipher.PubKey][]routing.Path`

**Review by evanlinjin — CHANGES_REQUESTED** [review]
Good work. However, how can we test this?

**Comment by ivcosla** [comment]
> Good work. However, how can we test this?

We can run integration tests, no matter which one as long as we test that routes are being created. They are working on my tests. Basically run the following:
 1) checkout both repositories (skywire and skywire-services) to `feature/route-finder-single-route`
 2) on skywire-services run `make integration-build`; `make integration-run-generic`; `make integration-startup`
 3) run `curl --data  {'"recipient":"'$PK_A'", "message":"Hello Joe!"}' -X POST  $CHAT_C`, if such message arrives to visor_a it's succesful 

For unit testing the API however I think it would be better to write them on the server side.

**Comment by evanlinjin** [comment]
@nkryuchkov please give this a review and see if we can merge this into your work so far. Thanks!

File: pkg/routing/route.go [hunk 2]
```diff
@@ -20,6 +21,41 @@ func (h Hop) String() string {
...
+	b1, err := p[0].MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	b2, err := p[1].MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	res := bytes.NewBuffer(b1)
+	res.WriteString(":") // nolint
```
Comment (nkryuchkov) [inline pkg/routing/route.go:38]: Although both `WriteString` and `Write` of `bytes.Buffer` always return `nil` as an error, I would explicitly check for an error here and below.

**Review by nkryuchkov — APPROVED** [review]
Great job!

**Comment by nkryuchkov** [comment]
@evanlinjin I've already merged it into my work

**Comment by evanlinjin** [comment]
> @evanlinjin I've already merged it into my work

Should we close this PR then?

**Comment by nkryuchkov** [comment]
@evanlinjin If it's to be merged in my work only, then yes.

**Comment by evanlinjin** [comment]
Already merged into #562

**Pull request closed without merging**